Added test cases for Style

wodeni opened the pull request:
Built a test suite for Style, which goes through all files in `TOP/src/sty` and attempts to parse them all. Each file is a separate test case. If any failure, the error message from the parser will be displayed (indentation is a bit off). @hypotext let me know if you think this is okay. If so, I’ll apply the same strategy to Substance tests.

Also, when I ran `sudo cabal test` the output has no color, but when I execute the binary directly in `dist/build/penrose-testsuite` the output is colored. @hypotext any ideas why?

Note that I had a lot of troubles with the dependencies... because I didn't know the test suite has a different set of dependencies in `penrose.cabal`. Maybe put that into the wiki as well?

hypotext commented:
Thanks for writing this! It's a great start. Some comments:

- When I do `stack build; stack test` in `TOP` the tests run and the output is colored. (`cabal` fails for me so I switched to stack)§.
- So far I see "15 out of 51 tests failed (2.59s)". Looks like some of the tests are failing on instances of old Style files, and others fail for reasons I don't see immediately. Would be good to clean up old files and resolve the current parse failures.

I'd suggest adding a few of the following kinds of tests, where some of them are more relevant to `getDictAndFns` (which is just as important as the parser):

- Test that the correct dictionaries and functions are being generated for specific features  (pattern-matching, Cartesian products, dot-accesses, etc.)
- Failure tests: make sure the parser or dictionary generator fails on inputs that should fail but might not be obvious, like `Set x, y { ... }`
- Some regression tests with golden files: if someone changes the parser, it would be helpful if it is immediately obvious which programs are being parsed differently. 

(You don't need to add a ton of these, since we might change things that might make certain tests obsolete, but do try and test the "core" features.)

hypotext commented:
Should I merge this?

wodeni commented:
I'll add test for Substance too. And then we merge. Is that okay?

On Thu, Feb 8, 2018 at 00:10 Katherine Ye <notifications@github.com> wrote:

> Should I merge this?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/penrose/penrose/pull/48#issuecomment-364004298>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ALMjxox7AdAqNcdQwhsPqc5za4ISLn7kks5tSoGmgaJpZM4RxMGr>
> .
>
-- 
Wode "Nimo" Ni
Class of 2018
School of Engineering and Applied Science
Columbia University
UNI: wn2155

hypotext commented:
Sounds good; let me know.

wodeni commented:
Added similar test cases for Substance and removed old Substance and Style files. Looks like Style will be changing quite a bit. Let me know if you would still like to have more complicated tests

hypotext commented:
When I check out this branch and do `stack build`, it seems to unregister every dependency and reinstall all of them (39). Do you happen to know why that's happening?

wodeni commented on stack.yaml:
```diff
@@ -1,8 +1,12 @@
+allow-newer: true
 packages:
 - .
 extra-deps:
```
Might be this thing I added causing the problem. Was playing around with `stack` earlier and accidentally pushed my local version. Sorry about that!

hypotext replied:
Ok, I'll remove this line when I merge my branch into master (unless you think it's better to keep it)

wodeni commented:
Feel free to do so. I'm not actively using stack anyway.
On Thu, Feb 22, 2018 at 12:46 Katherine Ye <notifications@github.com> wrote:

> *@hypotext* commented on this pull request.
> ------------------------------
>
> In stack.yaml:
>
> > @@ -1,8 +1,12 @@
>  flags: {}
>  extra-package-dbs: []
> +allow-newer: true
>  packages:
>  - .
>  extra-deps:
>
> Ok, I'll remove this line when I merge my branch into master (unless you
> think it's better to keep it)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/penrose/penrose/pull/48#discussion_r170039111>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ALMjxnZXGDgwPWHgx7hBZywpAWGCZlskks5tXafNgaJpZM4RxMGr>
> .
>
-- 
Wode "Nimo" Ni
Class of 2018
School of Engineering and Applied Science
Columbia University
UNI: wn2155
